flyway/src/main/java/io/micronaut/flyway/graalvm/MicronautPathLocationScanner.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.flyway.graalvm;
+
+import io.micronaut.context.condition.Condition;
+import org.flywaydb.core.api.Location;
+import org.flywaydb.core.internal.resource.LoadableResource;
+import org.flywaydb.core.internal.resource.classpath.ClassPathResource;
+import org.flywaydb.core.internal.scanner.classpath.ResourceAndClassScanner;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This class is used in order to prevent Flyway from doing classpath scanning which is both slow and won't work in
+ * native mode.
+ *
+ * Forked from the Quarkus implementation.
```

Comment:
Add reference to class forked and if this is a fork license headers and original `@author` tags need to be retained. If it is not a fork and merely inspired then change the wording.